src/compiler/transformers/destructuring.ts

```diff
@@ -331,11 +331,14 @@ namespace ts {
                 location
             );
         }
-        else if (numElements !== 1 && (flattenContext.level < FlattenLevel.ObjectRest || numElements === 0)) {
+        else if (numElements !== 1 && (flattenContext.level < FlattenLevel.ObjectRest || numElements === 0)
+            || every(elements, (element) => isOmittedExpression(element))) {

```

Comment:
Can we just use `every(elements, isOmittedExpression)`? No need to introduce a new function here.